client/core/core.go

```diff
@@ -4558,16 +4558,22 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 		}(id, waiter)
 	}
 	c.waiterMtx.Unlock()
+
 	c.connMtx.RLock()
-	assets := make(assetMap)
+	conns := make([]*dexConnection, 0, len(c.conns))
 	for _, dc := range c.conns {
+		conns = append(conns, dc)
+	}
+	c.connMtx.RUnlock()
+	assets := make(assetMap)
+	for _, dc := range conns {
```

Comment:
Another case where I don't believe there is good reason why the connMtx needed to be held during tickAsset.  The map access just needs to be safe protected from concurrent writes, but I don't think it needs to be guarded during the trade ticks.